tsec-libsodium/src/main/scala/tsec/libsodium/authentication/OnetimeAuth.scala

```diff
@@ -0,0 +1,37 @@
+package tsec.libsodium.authentication
+
+import cats.effect.Sync
+import tsec.libsodium.ScalaSodium
+import tsec.libsodium.cipher.SodiumKey
+
+sealed trait OnetimeAuth
+
+sealed trait MessageTag
+
+object OnetimeAuth {
+
+  def generateKey[F[_]](implicit F: Sync[F], S: ScalaSodium): F[SodiumKey[OnetimeAuth]] = F.delay {
+    val key = SodiumKey[OnetimeAuth](new Array[Byte](ScalaSodium.crypto_onetimeauth_poly1305_KEYBYTES))
+
+    S.crypto_onetimeauth_keygen(key)
```

Comment:
This is ok,  but just for the record, this is just `randombytes_buf` lol. I've avoided using the `x_keygen` functions for this reason.

Reply:
Ok if @hrhino and travis are ok too, i would sqash my commits to a single one.